integration_tests/tests/flowmachine_tests/test_available_dates.py

```diff
@@ -0,0 +1,91 @@
+import json
+import pytest
+
+from flowmachine.core.available_dates import AvailableDates
+
+
+@pytest.mark.parametrize(
+    "event_types, expected_result",
+    [
+        (
+            ["calls"],
+            [
+                {
+                    "event_type": "calls",
+                    "dates": [
+                        "2016-01-01",
+                        "2016-01-02",
+                        "2016-01-03",
+                        "2016-01-04",
+                        "2016-01-05",
+                        "2016-01-06",
+                        "2016-01-07",
```

Comment:
Where is the 8th?

Reply:
Ohhhhhhh it's a different month...suppose this is purposeful in the test data?

Reply:
Oh... I didn't even realise the different month. 😂 I simply copied the output from `flowdb_testdata`.

And yes, the hard-coded test data only contains the 1st-7th: https://github.com/Flowminder/FlowKit/tree/master/flowdb/testdata/test_data/data/records/calls

Also, the table for `2016-01-09` is empty so I presume it's added manually somewhere during startup?

Reply:
Ohhhhhhhhhhhh so this will tell me that a date is available even if it's an empty table? That's not good...Was there not some strictness setting for the original `available_dates`?

Reply:
Yes, but this was not implemented in SQL but rather as a Python add-on. I'm not sure this is immediately straightforward to port to the current machinery in which queries are exposed. Can have a think about it but it will need to be on Monday, sorry.

Reply:
The empty table is actually there specifically to be able to test the available_dates method on Connection. (Not a great way to do it)